tests/index.ts

```diff
@@ -1,9 +1,115 @@
 import { assert } from 'chai';
-import ApolloFetcher from '../src/index.js';
+import * as sinon from 'sinon';
+import HttpFetcher from '../src/httpFetcher';
+import { HttpObservable } from '../src/httpFetcher';
+import { State } from '../src/types';
+import gql from 'graphql-tag';
+import * as fetchMock from 'fetch-mock';
 
-describe('mockNetworkInterfaceWithSchema', () => {
-  it('can run a test', () => {
-    const apolloFetcher = new ApolloFetcher();
-    assert(apolloFetcher.canTest());
+const sampleQuery = gql`
+query SampleQuery {
+  stub{
+    id
+  }
+}
+`;
+
+describe('HttpFetcher', () => {
+  it('does not need any constructor arguments', () => {
+    assert.doesNotThrow(() => new HttpFetcher());
+  });
+
+  it('calls next and then complete', (done) => {
+    const next = sinon.spy();
+    const data = {hello: 'world'};
+    fetchMock.get('*', data);
+    const fetcher = new HttpFetcher('', fetch);
```

Comment:
We should make the `fetch` argument optional. We'll add a separate test for adding the argument later, which can use `sinon.spy` to check that the function is getting called.